services/src/main/java/org/keycloak/authentication/requiredactions/RecoveryAuthnCodesAction.java

```diff
@@ -98,6 +98,37 @@ public void processAction(RequiredActionContext reqActionContext) {
         generatedAtTime = Long.parseLong(httpReqParamsMap.getFirst(FIELD_GENERATED_AT_HIDDEN));
         generatedUserLabel = httpReqParamsMap.getFirst(FIELD_USER_LABEL_HIDDEN);
 
+        // Retrieve the initially generated recovery codes from the authentication session
+        String storedCodes = reqActionContext.getAuthenticationSession().getAuthNote("initialGeneratedCodes");
```

Comment:
Minor: Is it possible to introduce some constant (maybe in class `RecoveryAuthnCodesAction` which already contains few constants) and make sure that both places use this constant?